code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -178,7 +181,21 @@
 	R.module = RM
 	R.update_module_innate()
 	RM.rebuild_modules()
-	INVOKE_ASYNC(RM, .proc/do_transform_animation)
+
+	if(RM.use_old_animation && RM.old_animation_length)
+		R.icon = 'icons/hippie/mob/robot_transformations.dmi'
+		R.icon_state = RM.cyborg_base_icon
+		world << "[R.icon_state]"
+		R.dir = SOUTH
+		R.notransform = TRUE
+		flick(R.icon_state, R)
+		sleep(RM.old_animation_length+1)

```

Comment:
Please use the `addtimer` and `callback` system to do this instead of using spawns, you can find more information about this [here](https://github.com/HippieStation/HippieStation13/blob/master/code/datums/callback.dm)